src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -541,12 +593,18 @@ def _bracket_sensitive_look_ahead_match(cls, segments, matchers, parse_context):
                 # Now check have we closed all our brackets?
                 if bracket_stack:
                     # No we haven't.
-                    # TODO: Format this better
-                    raise SQLParseError(
-                        "Couldn't find closing bracket for opening bracket.",
-                        segment=bracket_stack.pop(),
+                    # Check that the unclosed brackets are definite
+                    definite_bracket_stack = list(
+                        filter(lambda b: b[1], zip(bracket_stack, definite_stack))
                     )
 
+                    # TODO: Format this better
+                    if definite_bracket_stack:

```

Comment:
I think this could be much more concise if we've combined `bracket_stack` and `definite_stack` as per my comment further up. Then this should actually just be:

NB: this is pseudocode, so won't actually work but it gets at the intent and avoids the filter and lambda expressions and makes this more readable.
```python
if any(bracket["definite"] for bracket in bracket_stack):
    raise SQLParseError("...")
```

Reply:
Done!